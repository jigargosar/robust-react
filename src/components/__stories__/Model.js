import {action} from '@storybook/addon-actions'
import {storiesOf} from '@storybook/react'
import {centerPaper} from '../../centerPaper'
import {h} from '../../hyper-script'
import {createFakeModels} from '../__fixtures__/Model.fixture'
import {ModelList} from '../Model'

const story = storiesOf('Components | ModelList', module).addDecorator(
  centerPaper,
)

story.add('basic', () =>
  h(ModelList, {
    models: createFakeModels(),
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
story.add('with primaryRenderer', () =>
  h(ModelList, {
    models: createFakeModels(),
    primaryRenderer: model => `${model.name} (id:${model.id})`,
    onClick: model => event => action('onClick')(model, event.type),
  }),
)
