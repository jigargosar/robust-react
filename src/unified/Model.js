import {ListItemText, MenuItem} from 'material-ui'
import {storiesOf} from '../facade'
import {h} from '../hyper-script'
import {centerPaper} from './story-decorators/centerPaper'
// import 'dom-testing-library/extend-expect'
// import {render} from 'react-testing-library'
// import {describe, expect, it, specs, storiesOf} from '../../../../facade'

const ModelListItem = ({model}) => {
  return h(MenuItem, [h(ListItemText, model.name)])
}

// const createFakeModels = () => {
//   const chance = Chance(11)
//   return R.times(
//     (idx: number) => ({id: idx, name: chance.country({full: true})}),
//     3,
//   )
// }

storiesOf('Unified | Model', module)
  .addDecorator(centerPaper)
  .add('ListItem', () => {
    return h(ModelListItem, {model: {id: 0, name: 'Foo bar'}})
  })
