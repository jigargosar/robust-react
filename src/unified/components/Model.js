import Chance from 'chance'
import {List, ListItemText, MenuItem} from 'material-ui'
import {head, map, propOr, times} from 'ramda'
import {storiesOf} from '../../facade'
import {h} from '../../hyper-script'
import {centerPaper} from './story-decorators/centerPaper'
// import 'dom-testing-library/extend-expect'
// import {render} from 'react-testing-library'
// import {describe, expect, it, specs, storiesOf} from '../../../../facade'

const story = storiesOf('Unified | Model', module).addDecorator(centerPaper)

const createFakes = () => {
  const chance = Chance(11)
  return times(idx => ({id: idx, name: chance.country({full: true})}), 3)
}

const ModelListItem = ({
  model,
  primary = propOr('<primary or name not found>', 'name'),
  onClick = () => () => {},
}) => h(MenuItem, {onClick: onClick(model)}, [h(ListItemText, primary(model))])

story.add('ListItem', () => h(ModelListItem, {model: head(createFakes())}))

const ModelList = ({models, ...other}) =>
  h(List, map(model => h(ModelListItem, {model, ...other}), models))

story.add('List', () => h(ModelList, {models: createFakes()}))

export {ModelList}
