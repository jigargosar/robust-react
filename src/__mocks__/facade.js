// import {render} from 'react-testing-library'
import noCrashSnapshot from '../components/__tests__/helpers/noCrashSnapshot'

const currentEnv = jasmine.currentEnv_

export const describe = currentEnv.describe
export const it = currentEnv.it
export const beforeEach = currentEnv.beforeEach
export const afterEach = currentEnv.afterEach
export const xit = currentEnv.xit
export const xdescribe = currentEnv.xdescribe
export const fit = currentEnv.fit
export const after = () => {}
export const before = () => {}
export const expect = global.expect

export const action = () => {}
export const linkTo = () => {}
export const specs = spec => spec()

// export const snapshot = (name, story) => {
//   it(name, () => {
//     const {container} = render(story)
//     expect(container.firstChild).toMatchSnapshot()
//   })
// }

export const storiesOf = kind => ({
  add(name, func) {
    describe(kind, () => {
      describe(name, () => {
        it('should render without crashing', () => {
          func()
        })

        it('should match snapshot', () => {
          // const {container} = render(func())
          // expect(container.firstChild).toMatchSnapshot()
          noCrashSnapshot(func)
        })
      })
    })

    return this
  },
  addDecorator() {
    return this
  },
})
