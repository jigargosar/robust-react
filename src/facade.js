import expect_ from 'expect'

expect_.extend({
  toMatchSnapshot: () => ({message: 'Fake Snapshot For Storybook', pass: true}),
})

export {default as expect} from 'expect'

export {default as jest} from 'jest-mock'
export {
  after,
  afterEach,
  before,
  beforeEach,
  describe,
  fit,
  it,
  specs,
  xdescribe,
  xit,
} from 'storybook-addon-specifications'

export {storiesOf} from '@storybook/react'
export {action} from '@storybook/addon-actions'
export {linkTo} from '@storybook/addon-links'
