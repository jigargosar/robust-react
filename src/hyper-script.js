import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
