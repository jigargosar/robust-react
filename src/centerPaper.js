import {Paper} from 'material-ui'
import {div, h} from './hyper-script'

const centerPaper = story =>
  div(
    {
      style: {
        padding: '16px 0',
        backgroundColor: '#ddd',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      },
    },
    [div({style: {flex: 1, maxWidth: 300}}, [h(Paper, [story()])])],
  )

export {centerPaper}
