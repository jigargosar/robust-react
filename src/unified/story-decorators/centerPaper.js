import {Paper} from 'material-ui'
import {storiesOf} from '../../facade'
import {div, h} from '../../hyper-script'

const centerPaper = story =>
  div(
    {
      style: {
        padding: '16px 0',
        backgroundColor: '#ddd',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      },
    },
    [div({style: {flex: 1, maxWidth: 300}}, [h(Paper, [story()])])],
  )

storiesOf('Unified | StoryDecorators', module).add('centerPaper', () => {
  return centerPaper(() => div('Story'))
})

export {centerPaper}
