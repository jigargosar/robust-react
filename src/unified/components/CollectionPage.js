import Chance from 'chance'
import {times} from 'ramda'
import {action, storiesOf} from '../../facade'
import {h} from '../../hyper-script'
import {ModelList} from './Model'
import {centerPaper} from './story-decorators/centerPaper'
// import 'dom-testing-library/extend-expect'
// import {render} from 'react-testing-library'
// import {describe, expect, it, specs, storiesOf} from '../../../../facade'

const chance = new Chance(11)

const createCollection = id => ({
  id,
  name: chance.country({full: true}),
  itemCount: chance.integer({min: 3, max: 10}),
})

const createFakes = () => {
  return times(createCollection, 3)
}

const story = storiesOf('Unified | Page', module).addDecorator(centerPaper)

const Collection = ({models, onClick}) =>
  h(ModelList, {
    models,
    primary: c => `${c.name} (${c.itemCount})`,
    onClick,
  })

story.add('Collection', () =>
  h(Collection, {
    models: createFakes(),
    onClick: collection => e =>
      action('collectionClicked')(collection.name, e.type),
  }),
)
