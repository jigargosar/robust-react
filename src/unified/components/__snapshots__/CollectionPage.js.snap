// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unified | Model List should match snapshot 1`] = `
<ul
  class="MuiList-root MuiList-padding"
>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
      >
        Ceuta & Melilla
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </li>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
      >
        Bahamas
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </li>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
      >
        American Samoa
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </li>
</ul>
`;

exports[`Unified | Model ListItem should match snapshot 1`] = `
<li
  class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <div
    class="MuiListItemText-root"
  >
    <h3
      class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
    >
      Ceuta & Melilla
    </h3>
  </div>
  <span
    class="MuiTouchRipple-root"
  />
</li>
`;

exports[`Unified | Page Collection should match snapshot 1`] = `
<ul
  class="MuiList-root MuiList-padding"
>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
      >
        Romania (10)
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </li>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
      >
        Iran (3)
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </li>
  <li
    class="MuiButtonBase-root MuiListItem-root MuiListItem-default MuiListItem-gutters MuiListItem-button MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <div
      class="MuiListItemText-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-subheading MuiListItemText-primary"
      >
        Laos (4)
      </h3>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </li>
</ul>
`;

exports[`Unified | StoryDecorators centerPaper should match snapshot 1`] = `
<div
  style="padding: 16px 0px; background-color: rgb(221, 221, 221); display: flex;"
>
  <div
    style="max-width: 300px;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div>
        Story
      </div>
    </div>
  </div>
</div>
`;
